database/mongodb/mongodb_test.go

```diff
@@ -104,3 +107,111 @@ func TestWithAuth(t *testing.T) {
 			}
 		})
 }
+
+func TestTransaction(t *testing.T) {
+	versionsSupportedTransactions := []mt.Version{
+		{
+			Image: "mongo:4",
+			Cmd: []string{
+				"/bin/bash",
+				"-c",
+				"mongod --fork --logpath /data/log.log --bind_ip_all --replSet rs0 && mongo --eval 'rs.initiate()' && tail -f /data/log.log",
```

Comment:
Does `mongo` wait for `mongod` to be ready while connecting? If not, there will probably be a race condition between `mongod` initializing and the `mongo --eval` connecting to `mongod` and running `rs.initiate()`.

Likewise, we need to use events to trigger actions instead of arbitrary wait times.

Reply:
Yes, `mongo` are waiting for initialization of `mongod`